pbparam/cost_functions/rmse.py

```diff
@@ -40,22 +40,26 @@ def evaluate(self, y_sim, y_data, sd=None):
         sd : array or list, optional
             contains the standard deviation of the data points, used for weighting
             the cost function. This variable will NOT be used in RMSE. Default is None.
+        weights : array or list, optional
+             contains custom weights for each data point. Size of weights must be
+             equal to size of reference data points!
 
         Returns
         -------
         RMSE : array
             Calculated RMSE for given inputs.
         """
-        # Ensure y_sim and y_data are lists
+        # Ensure y_sim, y_data adn weights are lists
         y_sim = y_sim if isinstance(y_sim, list) else [y_sim]
         y_data = y_data if isinstance(y_data, list) else [y_data]
+        weights = weights if isinstance(weights, list) else [weights]
 
         rmse = 0
 
         # Iterate over each simulation and data point
-        for sim, data in zip(y_sim, y_data):
+        for sim, data, weight in zip(y_sim, y_data, weights):
             # Calculate the error and normalize by the mean of the data
-            err = (sim - data) / np.nanmean(data)
+            err = ((sim - data) * weight) / np.nanmean(data)
```

Comment:
If we add weights, I think we should remove the division by `np.nanmean(data)`. This should be captured by the weights, which we set to ` 1 / np.nanmean(data)` if the weights argument is set to be `"mean data"` (or something like that). This ties to my point later on about the default value of `weights`.